experiments/two_stage/two_stage_config.py

```diff
@@ -0,0 +1,354 @@
+from levanter.tracker.wandb import WandbConfig
+from levanter.trainer import TrainerConfig
+
+from experiments.defaults import _prepare_data_config
+from experiments.evals.task_configs import convert_to_levanter_task_config
+from experiments.two_stage.data import data_dict
+from experiments.two_stage.models import model_dict
+from marin.evaluation.evaluation_config import EvalTaskConfig
+from marin.execution.executor import ExecutorStep, this_output_path
+from marin.processing.tokenize.data_configs import LMMixtureDatasetConfig, lm_varying_mixture_data_config
+from marin.training.training import PodConfig, TrainLmOnPodConfig, run_levanter_train_lm
+
+
+@dataclass
+class TwoStageConfig:
+    """Configuration for two-stage training."""
+
+    ### Data quantity
+    rare_data_name: str
+    common_data_name: str
+
+    rare_fraction: float = 0.01
+    rare_data_epochs: int = 1
+
+    ### Data schedule
+    # Can specify any 2 of the following 3 since the 3rd is implied.
+    # If you specify all three, will ensure they are consistent.
+    replay_ratio: float | None = None
+    rare_stage2_allocation: float | None = None
+    stage2_duration: float | None = None
+    data_seed: int = 42
+
+    ### Trainer config
+    num_train_steps: int = 1000
+    train_batch_size: int | None = 1024
+    steps_per_eval: int | None = None
+    wandb_project_name: str = "suhas-two-stage"
+    wandb_additional_tags: list[str] = field(default_factory=list)
+    steps_to_export_list: list[int] = field(default_factory=list)
+    nametag: str = ""
+
+    ### Hardware
+    tpu_type: str = "v4-128"
+    node_count: int = 1
+
+    ### Optimizer config
+    lr_schedule: str = "cosine"
+    lr: float = 3e-3
+    lr_cooldown_duration: float | None = None
+    weight_decay: float = 0.1
+    min_lr_ratio: float = 0.0
+
+    ### Model config
+    model_name: str = "150m4k"
+    initialize_from_checkpoint_path: str | None = None
+    initialize_from_hf: str | None = None
+
+    ### Eval config
+    eval_harness_tasks: list[EvalTaskConfig] | None = None
+    eval_harness_steps: int | None = None
+
+    def __post_init__(self):
+        self.model_config = model_dict[self.model_name]
+
+        self.set_data_schedule_params()
+
+        assert (
+            self.initialize_from_checkpoint_path is None or self.initialize_from_hf is None
+        ), "Cannot specify both initialize_from_checkpoint_path and initialize_from_hf"
+
+        if self.steps_per_eval is None:
+            self.steps_per_eval = self.num_train_steps // 20
+
+        self.steps_per_eval = min(self.steps_per_eval, self.num_train_steps // 2)
+
+        if self.eval_harness_steps is None and self.eval_harness_tasks is not None:
+            self.eval_harness_steps = self.num_train_steps // 4
+
+        self.rare_data = data_dict[self.rare_data_name]
+        self.common_data = data_dict[self.common_data_name]
+
+    def set_data_schedule_params(self):
+        """
+        ┌──────────────────────────┬──────────┐
```

Comment:
great ascii figure, but somewhere you need to give a high level explanation of what you're doign